src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2063,6 +2063,19 @@ MavlinkReceiver::handle_message_manual_control(mavlink_message_t *msg)
 	// For backwards compatibility at the moment interpret throttle in range [0,1000]
 	manual_control_setpoint.throttle = ((mavlink_manual_control.z / 1000.f) * 2.f) - 1.f;
 	manual_control_setpoint.yaw = mavlink_manual_control.r / 1000.f;
+
+	if (mavlink_manual_control.enabled_extensions & (1u << 2)) { manual_control_setpoint.aux1 = mavlink_manual_control.aux1; }
```

Comment:
```suggestion
	if (mavlink_manual_control.enabled_extensions & (1u << 2)) { manual_control_setpoint.aux1 = mavlink_manual_control.aux1;
	} else {
		manual_control_setpoint.aux1 = NAN;
	}
	
```

Reply:
That's exactly where I was going but first I need to make sure that's documented and handled appropriately. Then I feel confirmed and will do that 👍

Reply:
I realized I have to clarify my wording here:
Until now any RC channel not being mapped/available results in the corresponding `manual_control_setpoint` field being set to zero. See for example here: https://github.com/PX4/PX4-Autopilot/blob/ca9948a84d7b6aa8d66a8abeed001ec15a2d587e/src/modules/rc_update/rc_update.cpp#L275

This is neither explicitly documented nor does it allow the client to detect it's not available. So I'm happy to change this but I would not make it part of this pull request because it will be a lot of changes requiring extensive testing and only the suggested else cases need to be added for the functionality added in this pr.